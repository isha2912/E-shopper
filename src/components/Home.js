import React from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import Product from './Product';
import './Home.css';

const Home = (props) => {
  const { products, onIncrement, onDecrement } = props;
  return (
    <div className="all-products">
      {products.map((eachProduct) => (
        <Product
          key={eachProduct.id}
          product={eachProduct}
          onIncrement={() => onIncrement(eachProduct.id)}
          onDecrement={() => onDecrement(eachProduct.id)}
        />
      ))}
    </div>

  );
};

export default withRouter(Home);
Home.propTypes = {
  products: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      price: PropTypes.number,
      count: PropTypes.number,
      image: PropTypes.string,
      name: PropTypes.string,
    }).isRequired,
  ).isRequired,
  onDecrement: PropTypes.func.isRequired,
  onIncrement: PropTypes.func.isRequired,
};
