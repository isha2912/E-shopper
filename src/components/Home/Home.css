.all-products {
  display: "flex";
  display: flex;
  flex-direction: row;
  text-align: center;
  flex-wrap: wrap;
}

.table2 {
  padding: 2rem;
  margin-top: 1rem;
  align-self: center;
  height: 15em;
  width: 80rem;
  border-collapse: collapse;
}
