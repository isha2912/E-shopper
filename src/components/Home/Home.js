import React from 'react';
import PropTypes from 'prop-types';
import Product from '../Product/Product';
import './Home.css';

const Home = ({ filteredProducts, onIncrement, onDecrement }) => (
  <div>
    {Object.keys(filteredProducts).map((category) => (
      <div key={category} className="row">
        <h1>{category}</h1>
        <div className="all-products">
          {filteredProducts[category].map((product) => (
            <Product
              key={product.id}
              product={product}
              onIncrement={() => onIncrement(product.id, category)}
              onDecrement={() => onDecrement(product.id, category)}
            />
          ))}
        </div>
      </div>
    ))}
  </div>
);
export default Home;

Home.propTypes = {
  filteredProducts: PropTypes.shape({
    category: PropTypes.arrayOf(
      PropTypes.shape({
        stock: PropTypes.number.isRequired,
        count: PropTypes.number.isRequired,
        price: PropTypes.number.isRequired,
        name: PropTypes.string.isRequired,
      }),
    ),
  }).isRequired,
};
Home.propTypes = {
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
};
