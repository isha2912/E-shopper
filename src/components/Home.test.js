/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { screen, render } from '@testing-library/react';
import Home from './Home';

describe('Home page contains product components', () => {
  const mockProps = {
    filteredProducts: {
      fruits: [{
        id: 1,
        name: 'apple',
        price: 20,
        quantity: 1,
        count: 2,
      },
      ],
    },
    onIncrement: (() => {}),
    onDecrement: (() => {}),
  };

  afterEach(() => (jest.clearAllMocks()));
  test('Should display all product componenets', () => {
    const container = render(<Home {...mockProps} />);
    expect({ container }).toMatchSnapshot();
  });

  test('should display products', () => {
    render(<Home {...mockProps} />);
    screen.getByText('apple');
    screen.getByText('1 Kg');
    screen.getByText('MRP 20');
    screen.getByText('+');
    screen.getByText('-');
    screen.getByText('2 in Basket');
  });
});
