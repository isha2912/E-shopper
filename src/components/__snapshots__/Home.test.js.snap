// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page contains product components Should display all product componenets 1`] = `
Object {
  "container": Object {
    "asFragment": [Function],
    "baseElement": <body>
      <div>
        <div>
          <div
            class="row"
          >
            <h1>
              fruits
            </h1>
            <div
              class="all-products"
            >
              <div
                class="product-container"
              >
                <img
                  alt="fruit"
                  class="img-product"
                  src="assets/banana.png"
                />
                <div
                  class="product-name"
                  data-testid="card-img"
                >
                  apple
                </div>
                <div
                  class="product-quantity"
                >
                  1
                   
                  Kg
                </div>
                <div
                  class="product-price"
                >
                  MRP
                   
                  20
                </div>
                <div
                  class="counter"
                >
                  <button
                    class="button-decrement"
                    type="button"
                  >
                    -
                  </button>
                  <div
                    class="in-basket"
                  >
                    2
                     
                    in Basket
                  </div>
                  <button
                    class="button-increment"
                    type="button"
                  >
                    +
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </body>,
    "container": <div>
      <div>
        <div
          class="row"
        >
          <h1>
            fruits
          </h1>
          <div
            class="all-products"
          >
            <div
              class="product-container"
            >
              <img
                alt="fruit"
                class="img-product"
                src="assets/banana.png"
              />
              <div
                class="product-name"
                data-testid="card-img"
              >
                apple
              </div>
              <div
                class="product-quantity"
              >
                1
                 
                Kg
              </div>
              <div
                class="product-price"
              >
                MRP
                 
                20
              </div>
              <div
                class="counter"
              >
                <button
                  class="button-decrement"
                  type="button"
                >
                  -
                </button>
                <div
                  class="in-basket"
                >
                  2
                   
                  in Basket
                </div>
                <button
                  class="button-increment"
                  type="button"
                >
                  +
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>,
    "debug": [Function],
    "findAllByAltText": [Function],
    "findAllByDisplayValue": [Function],
    "findAllByLabelText": [Function],
    "findAllByPlaceholderText": [Function],
    "findAllByRole": [Function],
    "findAllByTestId": [Function],
    "findAllByText": [Function],
    "findAllByTitle": [Function],
    "findByAltText": [Function],
    "findByDisplayValue": [Function],
    "findByLabelText": [Function],
    "findByPlaceholderText": [Function],
    "findByRole": [Function],
    "findByTestId": [Function],
    "findByText": [Function],
    "findByTitle": [Function],
    "getAllByAltText": [Function],
    "getAllByDisplayValue": [Function],
    "getAllByLabelText": [Function],
    "getAllByPlaceholderText": [Function],
    "getAllByRole": [Function],
    "getAllByTestId": [Function],
    "getAllByText": [Function],
    "getAllByTitle": [Function],
    "getByAltText": [Function],
    "getByDisplayValue": [Function],
    "getByLabelText": [Function],
    "getByPlaceholderText": [Function],
    "getByRole": [Function],
    "getByTestId": [Function],
    "getByText": [Function],
    "getByTitle": [Function],
    "queryAllByAltText": [Function],
    "queryAllByDisplayValue": [Function],
    "queryAllByLabelText": [Function],
    "queryAllByPlaceholderText": [Function],
    "queryAllByRole": [Function],
    "queryAllByTestId": [Function],
    "queryAllByText": [Function],
    "queryAllByTitle": [Function],
    "queryByAltText": [Function],
    "queryByDisplayValue": [Function],
    "queryByLabelText": [Function],
    "queryByPlaceholderText": [Function],
    "queryByRole": [Function],
    "queryByTestId": [Function],
    "queryByText": [Function],
    "queryByTitle": [Function],
    "rerender": [Function],
    "unmount": [Function],
  },
}
`;
